temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -712,20 +711,28 @@ private void processScheduleNexusOperation(
       long workflowTaskCompletedId) {
     attr = validateScheduleNexusOperation(attr);
     Endpoint endpoint = nexusEndpointStore.getEndpointByName(attr.getEndpoint());
-    StateMachine<StateMachines.NexusOperationData> operationStateMachine =
-        newNexusOperation(endpoint);
+    StateMachine<StateMachines.NexusOperationData> operation = newNexusOperation(endpoint);
     long scheduleEventId = ctx.getNextEventId();
-    nexusOperations.put(scheduleEventId, operationStateMachine);
+    nexusOperations.put(scheduleEventId, operation);
 
-    operationStateMachine.action(Action.INITIATE, ctx, attr, workflowTaskCompletedId);
+    operation.action(Action.INITIATE, ctx, attr, workflowTaskCompletedId);
+    // Use TIMEOUT_TYPE_SCHEDULE_TO_START for timing out an individual start request
```

Comment:
I see now why you're using schedule to start, that's a bit confusing since we'll add schedule to start support for the operation that's intended to time out from operation schedule time, not attempt schedule time.

Reply:
I'm not super familiar with the Java test server but the best way to timeout dispatching a task would be to put the timeout on the task itself but I don't know how that would interact with time skipping.

If you must use `ctx.addTimer`, consider using a `failNexusOperationAttempt` method to avoid overloading schedule to start.

Reply:
Make sure you also set the timeout on the task so the SDK can expose the task timeout to the operation handler.

Reply:
Yeah I don't think we can put the timer on the task itself. I will refactor this a bit to avoid overloading schedule to start.

Reply:
Isn't the deadline set on a header https://github.com/temporalio/temporal/blob/c86712f795c168ebe7e73cca0a68849afa33d5ae/service/matching/matching_engine.go#L1759?

Reply:
I am still using `ctx.addTimer` but I am reusing the `failNexusOperation` to handle attempt retries instead of having a separate method.